packages/ui/src/components/TermsOfUse/index.tsx

```diff
@@ -26,25 +27,28 @@ const TermsOfUse = ({ name, className, termsOfUse, isChecked, onChange }: Props)
   const prefix = t('sign_in.terms_agreement_prefix');
 
   return (
-    <div
-      className={classNames(styles.terms, className)}
-      onClick={() => {
-        onChange(!isChecked);
-      }}
-    >
-      <input disabled readOnly name={name} type="checkbox" checked={isChecked} />
-      <RadioButtonIcon checked={isChecked} className={styles.radioButton} />
-      <div className={styles.content}>
-        {prefix}
-        <TextLink
-          text="sign_in.terms_of_use"
-          href={termsOfUse.contentUrl}
-          type="secondary"
-          onClick={(event) => {
-            event.stopPropagation();
-          }}
-        />
+    <div className={className}>
+      <div
+        className={styles.terms}
+        onClick={() => {
+          onChange(!isChecked);
+        }}
+      >
+        <input disabled readOnly name={name} type="checkbox" checked={isChecked} />
+        <RadioButtonIcon checked={isChecked} className={styles.radioButton} />
+        <div className={styles.content}>
+          {prefix}
+          <TextLink
+            text="sign_in.terms_of_use"
+            href={termsOfUse.contentUrl}
+            type="secondary"
+            onClick={(event) => {
+              event.stopPropagation();
+            }}
```

Comment:
Is this onClick handler necessary?

Reply:
Since I make the whole parent wrapper clickable to toggle the radioButton status. By clicking on the link itself should not trigger the parent's onClick event right. Is there a better approach?

Reply:
Better leave a comment.

Reply:
added

Reply:
The whole sentence should be clickable except for the link